### PR TITLE
add config_name and data_files to datasetcard.md specs

### PR DESCRIPTION
**Pull request by lappemic**
As realised in #1222 the specs of the datasetcard are not up to date. This PR therefore updates the specs.

Let me know if this is sufficient or you would like to add more config examples.

Cc: @severo, @polinaeterna

### CODE REVIEW

**Review by julien-c — APPROVED** [review]
LGTM, thanks

Note that there's also a `config_names: list[str]` (i.e the previous property) but i don't think we need to document it

File: datasetcard.md [hunk 1]
```diff
@@ -33,9 +33,13 @@ task_ids:
...
-configs:  # Optional for datasets with multiple configurations like glue.
-- {config_0}  # Example for glue: sst2
-- {config_1}  # Example for glue: cola
+configs:  # Optional for datasets with multiple configurations.
+  - config_name: default
+    data_files:
+      - split: train
+        path: "data.csv"
+      - split: test
+        path: "holdout.csv"
```
Comment (polinaeterna) [inline datasetcard.md:42]:
```suggestion
configs:  # Optional. This can be used to pass additional parameters to the dataset loader, such as `data_files`, `data_dir`, and any builder-specific parameters  
  - config_name: {config_name_0}  # Example: default
    data_files:
      - split: {split_name_0}  # Example: train
        path: {file_path_0}  # Example: data.csv
      - split: {split_name_1}  # Example: test
        path: {file_path_1}   # Example: holdout.csv
   - config_name: {config_name_1}  # Example: ...
      ... #TODO
```
and then we can also add example, ideally for some real dataset with multiple configs, maybe copy a part from [nyu-mll/glue](https://huggingface.co/datasets/nyu-mll/glue/blob/main/README.md?code=true#L346)?

**Review by polinaeterna — CHANGES_REQUESTED** [review]
thank you a lot for working on this! 
I've left a suggestion to reformat it a bit to keep it aligned with the rest of the file.

Also I wonder if we should try to include all possible cases in these specs or keep it simple as it is now and refer to https://huggingface.co/docs/hub/datasets-manual-configuration instead? cc @severo  because there are a lot of parameters apart from `data_files` that can be passed, for example for `csv` builder one can specify a `sep`, for audio/imagefolder builders - `drop_metadata`/`drop_labels`, etc. there is also `data_dir` parameter, similar to `data_files`.
I personally would keep it simple but not sure cc @lhoestq @stevhliu

File: datasetcard.md (hunk 1 above)
Comment (lappemic) [reply datasetcard.md:42]: Looks clearer, thanks. 

About the example: How about using [ELI5 dataset card](https://huggingface.co/datasets/eli5/blob/main/README.md)? I come up because it is already used in [Dataset card creatino guide](https://huggingface.co/docs/hub/datasets-cards#dataset-card-creation-guide). I find it always easier when the same example is referenced. WDYT?

**Comment by stevhliu** [comment]
> I personally would keep it simple but not sure

Since this is just an example dataset card, we don't necessarily need to include all possible parameters. A simple example use case should suffice, with a link to more options (to allow users to explore more gradually if they'd like), as you suggested sounds great to me.

File: datasetcard.md [hunk 2]
```diff
@@ -33,9 +33,15 @@ task_ids:
...
-- {config_1}  # Example for glue: cola
+configs:  # Optional. This can be used to pass additional parameters to the dataset loader, such as `data_files`, `data_dir`, and any builder-specific parameters  
+  - config_name: {config_name_0}  # Example: default
+    data_files:
+      - split: {split_name_0}  # Example: train
+        path: {file_path_0}  # Example: data.csv
+      - split: {split_name_1}  # Example: test
+        path: {file_path_1}   # Example: holdout.csv
+   - config_name: {config_name_1}  # Example: ...
+      ... #TODO
```
Comment (lhoestq) [inline datasetcard.md:44]: (nit) smaller indentation
```suggestion
configs:  # Optional. This can be used to pass additional parameters to the dataset loader, such as `data_files`, `data_dir`, and any builder-specific parameters  
- config_name: {config_name_0}  # Example: default
  data_files:
  - split: {split_name_0}  # Example: train
    path: {file_path_0}  # Example: data.csv
  - split: {split_name_1}  # Example: test
    path: {file_path_1}   # Example: holdout.csv
- config_name: {config_name_1}  # Example: ...
  ... #TODO
```

**Review by lhoestq — APPROVED** [review]
thanks for the addition ! looks like we missed this update

> Also I wonder if we should try to include all possible cases in these specs or keep it simple as it is now and refer to https://huggingface.co/docs/hub/datasets-manual-configuration instead? cc @severo because there are a lot of parameters apart from data_files that can be passed, for example for csv builder one can specify a sep, for audio/imagefolder builders - drop_metadata/drop_labels, etc. there is also data_dir parameter, similar to data_files.
I personally would keep it simple but not sure cc @lhoestq @stevhliu

I think it's fine for now. We only have this [page](https://huggingface.co/docs/datasets/en/package_reference/loading_methods#from-files) for a full list of parameters, it would be cool to have a more readable page for this that we could link to from this spec file later

File: datasetcard.md (hunk 2 above)
Comment (julien-c) [reply datasetcard.md:44]: yes, and remove the #TODO at the end

File: datasetcard.md [hunk 3]
```diff
@@ -33,9 +33,14 @@ task_ids:
...
-- {config_0}  # Example for glue: sst2
-- {config_1}  # Example for glue: cola
+configs:  # Optional. This can be used to pass additional parameters to the dataset loader, such as `data_files`, `data_dir`, and any builder-specific parameters  
+- config_name: {config_name_0}  # Example: default
+  data_files:
+  - split: {split_name_0}  # Example: train
+    path: {file_path_0}  # Example: data.csv
+  - split: {split_name_1}  # Example: test
+    path: {file_path_1}   # Example: holdout.csv
+- config_name: {config_name_1}  # Example: ...
```
Comment (polinaeterna) [inline datasetcard.md:43]:
```suggestion
- config_name: {config_name_1}  # Example: processed
  data_files:
    - split: {split_name_3}  # Example: train
      path: {file_path_3}  # Example: data_processed.csv
```
let's complete this block for second config? and we can merge

**Review by polinaeterna — APPROVED** [review]
